growthbook/growthbook
# 895, 896, 897, 898 • Slack integration list view (w/ details), Add, Edit and Delete functionality

### Features and Changes

Adds a new page /integrations/slack for creating Slack integrations. The integrations right now are not functional. This UI is hidden and not available to users. Supporting the actual alerting will be done in an upcoming PR.

Adds support for creating, updating, and deleting Slack integrations.

Includes validation, which should disable the Create or Save button.

- Closes #895 
- Closes #896 
- Closes #897 
- Closes #898 


### Dependencies

n/a

### Testing

Navigate to http://localhost:3000/integrations/slack and perform the following operations:

- create a new Slack integration
- update an existing Slack integration
- delete a Slack integration

Play around with all the field types. You should be able to change the field values without any issue.


## Screenshots

### List view

<img width="1230" alt="image" src="https://user-images.githubusercontent.com/113377031/212447765-68de87e8-889d-495e-86ed-5b240e53b764.png">


<img width="1230" alt="image" src="https://user-images.githubusercontent.com/113377031/212447724-016c7822-d8a6-4da0-97d4-df9b46e548f0.png">

<img width="483" alt="image" src="https://user-images.githubusercontent.com/113377031/212447798-b37ed8a9-c0ed-4d16-b577-49c430ea6ad8.png">

<img width="477" alt="image" src="https://user-images.githubusercontent.com/113377031/212447791-110d17c8-7b90-41ef-a0bf-93a812e60e99.png">

Add button at the bottom of the list:
<img width="1231" alt="image" src="https://user-images.githubusercontent.com/113377031/213257270-52bc2297-c6e8-4db8-802c-92a3e4fc0b85.png">


#### Empty state

<img width="1108" alt="image" src="https://user-images.githubusercontent.com/113377031/212448756-a746dbef-2a94-496c-b762-86fae08322b9.png">

<img width="1102" alt="image" src="https://user-images.githubusercontent.com/113377031/212448763-525f01c1-a5c5-4566-b427-e2a42dc468eb.png">



### Create modal

<img width="532" alt="image" src="https://user-images.githubusercontent.com/113377031/212447867-dad12301-9161-4593-bfa7-3804c4fd76e0.png">

<img width="535" alt="image" src="https://user-images.githubusercontent.com/113377031/212447818-865a631a-b6be-47f0-a209-9dcb5f04527a.png">

### Edit modal

![CleanShot 2023-01-13 at 19 00 55@2x](https://user-images.githubusercontent.com/113377031/212447890-b99a430e-63f0-48b9-bc63-0757eb2daedb.png)

![CleanShot 2023-01-13 at 19 01 31@2x](https://user-images.githubusercontent.com/113377031/212447914-12a53b7c-1b75-46d8-b7a9-98b9fa65bcb1.png)


### Delete modal

<img width="540" alt="image" src="https://user-images.githubusercontent.com/113377031/212447934-1d04c1d2-e9ca-48e0-82b4-d7234f1b9ff7.png">
